core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -1017,4 +1053,16 @@ public FeaturesConfig setPlanWithTableNodePartitioning(boolean planWithTableNode
         this.planWithTableNodePartitioning = planWithTableNodePartitioning;
         return this;
     }
+
+    public JoinPushdownMode getJoinPushdownMode()
+    {
+        return joinPushdownMode;
+    }
+
+    @Config("optimizer.join-pushdown-enabled")
```

Comment:
Typically it would be: `s/-enabled/.enabled`
however in this case i think we can scratch the enabled part, if this becomes an enum:
```
optimizer.join-pushdown { ENABLED | DISABLED | AUTOMATIC }
```

or call it
```
optimizer.join-pushdown.mode
```

but i prefer my first proposal

Reply:
How would you see session property.

```
join_pushdown "ENABLED" "DISABLED", "AUTOMATIC"
```
will be fine?

Reply:
looks fine to me